### PR TITLE
Feature/cross section api

### PR DESCRIPTION
**Pull request by ryan-kipawa**
Includes #142

### CODE REVIEW

File: mikeio1d/cross_sections/cross_section_collection.py [hunk 1]
```diff
@@ -221,6 +221,11 @@ def __or__(self, other) -> CrossSectionCollection:
         else:
             super().__or__(other)
 
+    @property
+    def cross_section_data(self) -> CrossSectionData:
```
Comment (gedaskir) [inline mikeio1d/cross_sections/cross_section_collection.py:235]: I am thinking if we should do the same for Res1D, i.e., call it result_data. Or we could call here just data as well. But I think it makes sense to streamline.

File: mikeio1d/various.py [hunk 2]
```diff
@@ -99,3 +103,24 @@ def allow_nested_autocompletion_for_ipython(cls: Type):
     from IPython.core import guarded_eval
 
     guarded_eval.EVALUATION_POLICIES["limited"].allowed_getattr.add(cls)
+
+
+def open(file_name: str | Path, **kwargs) -> Res1D | Xns11:
+    """Open a file type supported by MIKE IO 1D file."""
+    # import here to avoid circular imports
+    from mikeio1d.res1d import Res1D
```
Comment (gedaskir) [inline mikeio1d/various.py:111]: OK, I see a problem here :-). Actually, it looks that various.py is not a good place for it, so I would just elevate the open method to a open.py.

**Comment by gedaskir** [comment]
> Includes #142

We ditch #142 ?

File: mikeio1d/cross_sections/cross_section_collection.py [hunk 3]
```diff
@@ -35,7 +34,7 @@
 TopoId = str
 
 
-class CrossSectionCollection(Dict[Tuple[LocationId, Chainage, TopoId], CrossSection]):
+class CrossSectionCollection(MutableMapping[Tuple[LocationId, Chainage, TopoId], CrossSection]):
```
[the remark this answers is not in the document]
Comment (ryan-kipawa) [reply mikeio1d/cross_sections/cross_section_collection.py:38]: This class isn't _truly_ a dictionary, and is more similar to a generic mapping. So the idea is to favor composition backed by a `dict`, while making sure we implement the right methods for it to act like a pythonic mapping.

File: mikeio1d/__init__.py [hunk 4]
```diff
@@ -53,6 +53,8 @@
 allow_nested_autocompletion_for_ipython(Res1D)
 allow_nested_autocompletion_for_ipython(Xns11)
 
+from .various import open as open_impl
+
 
 def open(file_name: str | Path, **kwargs) -> Res1D | Xns11:
```
[the remark this answers is not in the document]
Comment (ryan-kipawa) [reply mikeio1d/__init__.py:59]: You're right, actually there's no good reason I can think of 😛

File: mikeio1d/cross_sections/cross_section_collection.py (hunk 1 above)
Comment (ryan-kipawa) [reply mikeio1d/cross_sections/cross_section_collection.py:235]: Yeah I agree. I'll add a property for ResultData in Res1D as `result_data`

File: mikeio1d/various.py (hunk 2 above)
Comment (ryan-kipawa) [reply mikeio1d/various.py:111]: Yep not the best. I'll put it in `top_level.py` for now, then we can revisit the naming of that module 😄

File: mikeio1d/res1d.py [hunk 5]
```diff
@@ -563,6 +563,15 @@ def projection_string(self) -> str:
 
     # region deprecation
 
+    @property
+    def data(self) -> ResultData:
+        """Deprecated. Use 'result_data' instead."""
+        warnings.warn(
+            "The 'result_data' parameter will be deprecated in 1.0. Use 'data' instead.",
```
Comment (gedaskir) [inline mikeio1d/res1d.py:570]: Shouldn't it be the other way around?

File: mikeio1d/various.py [hunk 6]
```diff
@@ -6,16 +6,20 @@
 
 if TYPE_CHECKING:
     from typing import Type
+    from mikeio1d import Res1D
```
Comment (gedaskir) [inline mikeio1d/various.py:9]: I suggest to remove Res1D and Xns11 reference from various.py

**Review by gedaskir — APPROVED** [review]

**Pull request merged**